### ISSUE TITLE
payment_method access

### ISSUE THREAD

**Issue by ball1008** [stripe/stripe-terminal-android#57]
After creating payment intent, retrieving the intent on my server I call collectPaymentMethod which successfully returns the intent.  Next I want to confirm the intent which requires a payment_method from the intent.  We are not able to access the payment_method from the intent object.

**Comment by danwaters-stripe** [comment]
Hi,

It sounds like you're trying to to confirm the payment intent on the server, after calling `Terminal.collectPaymentMethod`. While we allow you to create payment intents on the server side, we don't support confirming them from the server. Instead, you can use `Terminal.processPayment` to confirm the intent.

**Comment by ball1008** [comment]
Thanks. I have tried confirming using Terminal.processPayment but it is not working.  I create  then retrieve my intent successfully and then call:

public void getPaymentMethod(PaymentIntent paymentIntent){
    Cancelable cancelable = Terminal.getInstance().collectPaymentMethod(paymentIntent,
            StripePayment.this,
            new PaymentIntentCallback() {
                @Override
                public void onSuccess(PaymentIntent paymentIntent) {
                    // Placeholder for processing paymentIntent
                    processPayment(paymentIntent);
                }

                @Override
                public void onFailure(TerminalException exception) {
                    // Placeholder for handling exception
                    Toast.makeText(_context, exception.getErrorMessage(), Toast.LENGTH_LONG).show();

                }
            });
}
This comes back with an intent that I then use to call:

private void processPayment(PaymentIntent paymentIntent) {
    Terminal.getInstance().processPayment(paymentIntent,
            new PaymentIntentCallback() {
                @Override
                public void onSuccess(PaymentIntent paymentIntent) {
                    // Placeholder for notifying your backend to capture paymentIntent.id
                    Toast.makeText(_context, paymentIntent.getId(), Toast.LENGTH_LONG).show();
                    capturePayment(paymentIntent);
                }

                @Override
                public void onFailure(TerminalException exception) {
                    // Placeholder for handling the exception
                    Toast.makeText(_context, "Error processing payment: " + exception.getLocalizedMessage(), Toast.LENGTH_LONG).show();

                }
            });
}
It seems that the payment method type is not created and added to the intent even though I get a message to insert, tap , or swipe and a message that says remove card.

Jen

**Comment by danwaters-stripe** [comment]
Interesting. Do you mind posting the logs that result from trying this? Make sure that you call `initTerminal` with `LogLevel.VERBOSE` so that we can get the full details of what's happening.

**Comment by ball1008** [comment]
Dan,

Here is the error from my logs:

09-13 17:28:11.894 18512-18512/com.foodmoodbackend.phillyGrilleTest D/StripeTerminal: class=Terminal message=collectPaymentMethod
09-13 17:28:11.895 18512-18686/com.foodmoodbackend.phillyGrilleTest E/StripeTerminal: UNEXPECTED_ERROR.UNEXPECTED_SDK_ERROR: Operation in progress wasn't completed before starting new operation
    com.stripe.stripeterminal.model.external.TerminalException
        at com.stripe.stripeterminal.TerminalSession$enqueueOperation$1.run(TerminalSession.kt:193)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:423)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
        at java.lang.Thread.run(Thread.java:818)
09-13 17:28:11.896 18512-18686/com.foodmoodbackend.phillyGrilleTest W/System.err: com.stripe.stripeterminal.model.external.TerminalException
09-13 17:28:11.896 18512-18686/com.foodmoodbackend.phillyGrilleTest W/System.err:     at com.stripe.stripeterminal.TerminalSession$enqueueOperation$1.run(TerminalSession.kt:193)
09-13 17:28:11.897 18512-18686/com.foodmoodbackend.phillyGrilleTest D/StripeTerminal: class=TerminalListenerProxy message=onPaymentStatusChange(WAITING_FOR_INPUT)
09-13 17:28:11.897 18512-18686/com.foodmoodbackend.phillyGrilleTest D/StripeTerminal: class=BbposBluetoothAdapter message=collectPaymentMethod

Jen

**Comment by ball1008** [comment]
Dan

With the new version I am not getting to the point where I confirm.  I'm getting Operation in progress wasn't completed before starting new operation.  Logs included in my prior comment.

**Comment by chander-stripe** [comment]
Closing out this stale issue. If you suspect a bug with the SDK, please don't hesitate to file a new issue. If you need further debugging help, please reach out to the Stripe technical support team. Thanks!
